### PR TITLE
Generate schema docs

### PR DESCRIPTION
**Pull request by SimonRichardson**
## Description of change

The following handles the facade schema with type and method
documentation. It's a copy of what already happens for the old
js-jujulib documentation creation, but instead, we lift this into juju so
other client libs can use it.

## QA steps

```sh
make rebuild-schema
cd scripts/schemadocs
make clean output.html
```

## Bug reference

https://bugs.launchpad.net/juju/+bug/1882571
https://bugs.launchpad.net/juju/+bug/1880634

### CODE REVIEW

**Comment by hpidcock** [comment]
!!build!!

File: generate/schemagen/schemagen.go [hunk 1]
```diff
@@ -5,11 +5,21 @@ package main
...
 	"os"
 
+	"github.com/juju/errors"
 	"github.com/juju/juju/apiserver"
+	"github.com/juju/juju/apiserver/common"
+	"github.com/juju/juju/apiserver/facade"
+	"github.com/juju/juju/core/permission"
 	"github.com/juju/juju/generate/schemagen/gen"
+	"github.com/juju/juju/state"
+	"github.com/juju/names/v4"
```
Comment (hpidcock) [inline generate/schemagen/schemagen.go:21]: Need to regroup these imports.

File: generate/schemagen/gen/generator.go [hunk 2]
```diff
@@ -3,49 +3,250 @@
...
+
+		pt, err := progType(pkg, objType.GoType())
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		doc, err := typeDocComment(pkg, pt)
+		if err != nil {
+			return nil, errors.Annotatef(err, "cannot get doc comment for %v", objType.GoType())
+		}
+		result[i].Description = doc
```
Comment (hpidcock) [inline generate/schemagen/gen/generator.go:95]: Would be good to trim leading and following whitespace.

**Comment by SimonRichardson** [comment]
$$merge$$

**Pull request merged**